### PR TITLE
refactor: handle errors in redeem retry

### PR DESCRIPTION
**Pull request by gregdhill**
Signed-off-by: Gregory Hill <gregorydhill@outlook.com>

### CODE REVIEW

File: vault/src/execution.rs [hunk 1]
```diff
@@ -188,16 +188,25 @@ impl Request {
...
+                    self.hash,
+                    H256Le::from_bytes_le(tx_metadata.txid.as_ref()),
+                    tx_metadata.proof.clone(),
+                    tx_metadata.raw_tx.clone(),
+                )
+            },
+            |result| match result {
+                Ok(ok) => Ok(ok),
+                Err(err) if err.is_commit_period_expired() => Err(RetryPolicy::Throw(err)),
+                Err(err) if err.is_rpc_error() => Err(RetryPolicy::Throw(err)),
```
Comment (sander2) [inline vault/src/execution.rs:206]: Why not retry on rpc error?

Comment (gregdhill) [reply vault/src/execution.rs:206]: Good point, I probably need to distinguish the errors. I was thinking just about the disconnect case.

**Review by sander2 — APPROVED** [review]

**Comment by sander2** [comment]
Maybe you can open an issue on the appropriate partity repo to get more specific errors?

**Comment by gregdhill** [comment]
https://github.com/paritytech/substrate/issues/8790

**Pull request merged**